### PR TITLE
Power Control Actions

### PR DESCRIPTION
**Pull request by dturn**
Powers: https://github.com/Shopify/genesis/pull/319

@dwradcliffe @dalehamel

### CODE REVIEW

File: lib/power_control.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def perform(item)
+    ipmi_ip = item['device']['ipmi_ip']
+
+    case item['payload']['desired_action']
+    when 'on'
+      ipmi_tool(ipmi_ip, 'chassis power on')
+    when 'off'
+      ipmi_tool(ipmi_ip, 'chassis power off')
+    when 'restart'
```
Comment (dalehamel) [inline lib/power_control.rb]: note: cycle will not work if the chassis is off

Comment (dturn) [inline lib/power_control.rb]: Would `reset` work instead? Or should we leave it as is

Comment (dalehamel) [inline lib/power_control.rb]: I think we can have it safely do 'power on' before any power cycle. This is what we do for zygote actions right now

Comment (dalehamel) [inline lib/power_control.rb]: We can also make zygote actions that do reboots use this power cycle command

**Comment by dalehamel** [comment]
looking good :)

**Comment by dwradcliffe** [comment]
👍

**Pull request merged**